meson.build

```diff
@@ -421,28 +421,34 @@ decl_headers = '''
 #include <uchar.h>
 #include <linux/ethtool.h>
 #include <linux/fib_rules.h>
-#include <linux/stat.h>
 #include <sys/stat.h>
 '''
-# FIXME: key_serial_t is only defined in keyutils.h, this is bound to fail
-# FIXME: these should use -D_GNU_SOURCE, since that is defined at build time
 
 foreach decl : ['char16_t',
                 'char32_t',
-                'key_serial_t',
                 'struct ethtool_link_settings',
                 'struct fib_rule_uid_range',
                 'struct statx',
                ]
 
         # We get -1 if the size cannot be determined
-        have = cc.sizeof(decl, prefix : decl_headers) > 0
+        have = cc.sizeof(decl, prefix : decl_headers, args : '-D_GNU_SOURCE') > 0
+
+        if decl == 'struct statx'
+                if have
+                        want_linux_stat_h = false
+                else
+                        have = cc.sizeof(decl,
+                                         prefix : decl_headers + '#include <linux/stat.h>',
```

Comment:
nitpick: Perhaps adding a newline there would be safer, in case decl_headers is later changed to check each struct/include pair individually and we have `'#include <sys/stat.h>' + '#include <linux/stat.h>'` which won't work...

Perhaps including `<linux/stat.h>` first would be slightly better, matches current behavior...

Does this work? `prefix : '#include <linux/stat.h>\n' + decl_headers`

Reply:
We know that `decl_headers` has a trailing newline, so this is OK. If it is changed, then this might need to change, but it'll be obvious. So I think think simplest form is good enough.

> Perhaps including <linux/stat.h> first would be slightly better, matches current behavior...

True, but I don't think matching current behaviour matters. It should be good enough to include the headers in any order, so again, the simplest form should be good enough.